ethereum/blockwatch/stack_test.go

```diff
@@ -0,0 +1,99 @@
+package blockwatch
+
+import (
+	"context"
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/0xProject/0x-mesh/common/types"
+	"github.com/0xProject/0x-mesh/db"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	miniHeaderOne = &types.MiniHeader{
+		Number:    big.NewInt(1),
+		Hash:      common.HexToHash("0x1"),
+		Parent:    common.HexToHash("0x0"),
+		Timestamp: time.Now().UTC(),
+	}
+	miniHeaderTwo = &types.MiniHeader{
+		Number:    big.NewInt(2),
+		Hash:      common.HexToHash("0x2"),
+		Parent:    common.HexToHash("0x1"),
+		Timestamp: time.Now().UTC(),
+	}
+)
+
+func newTestStack(t *testing.T, ctx context.Context) *Stack {
+	database, err := db.New(ctx, db.TestOptions())
+	require.NoError(t, err)
+	return NewStack(database)
+}
+
+func TestSimpleStackPushPeekPop(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	stack := newTestStack(t, ctx)
+
+	err := stack.Push(miniHeaderOne)
+	require.NoError(t, err)
+
+	expectedLen := 1
+	miniHeaders, err := stack.PeekAll()
```

Comment:
Can we check equality instead of length here?

Reply:
I didn't actually write these tests. I copied them over from Fabio's code. But you're right. I'll clean them up a bit.